### DISCUSSION TITLE
Second delivery to NISAR ADT on Raider

### DISCUSSION THREAD

**Discussion by dbekaert** [dbekaert/RAiDER#430] [General]
Pythonize the access to Raider for NISAR ADT to call within own codes without the CLI.

Example:
```
import pyaps3 as pa

pa.ECMWFdload(
        date_list2dload,
        hour,
        grib_dir,
        model=tropo_model,
        snwe=snwe,
        flist=grib_files2dload)

aps_obj = pa.PyAPS(
        grib_file,
        grib=tropo_model,
        Del=delay_type,
        dem=dem,
        inc=inc,
        lat=lat,
        lon=lon,
        mask=mask,
        verb=verbose)
```

To enable this we will modify key top functions (delay calculation, model download):
- [x] from RAiDER.processWM import prepareWeatherModel. This downloads and pre-processes.
- [x] from RAiDER.delay import main. We should change that to tropoDelay or so for more descriptive name

We will apply a 3-step incremental approach on this incremental delivery, with at the end a new minor release:
1. We will deliver the python entries without documentation -  aim for Friday 9 Dec
2. We will deliver the updated documentation for the entry points. We will be using doc strings for this. - aim for Friday 16 Dec
3. Do a new minor Raider release - aim for Friday 16 Dec

**Comment by jlmaurer** [comment]
Pythonization provided in #437. Documentation still to come.

**Comment by dbekaert** [comment]
# NISAR ADT RAiDER release update
Key new features are ([see release notes for details](https://github.com/dbekaert/RAiDER/releases/tag/v0.3.0)): 
- support for python library
- release of github IO documentation ([https://dbekaert.github.io/RAiDER/](https://dbekaert.github.io/RAiDER/)) 
- parsing of delay outputs in memory for  delay computation in python
-  slant delay computation on the cube

### RAiDER release
v0.3.0 (patch has been release v0.3.1)
### Installing
To get started, run the following lines in a terminal:
```
conda create --name RAiDER  -c conda-forge raider
conda activate RAiDER
```
Then
```
raider.py --help
```
### Notebook examples:
- Raider examples with configure file: https://github.com/dbekaert/RAiDER-docs/blob/main/notebooks/RAiDER_tutorial/RAiDER_tutorial.ipynb
- (NEW) Python Library demo: https://github.com/dbekaert/RAiDER-docs/blob/main/notebooks/RAiDER_tutorial/RAiDER_Python_access.ipynb

### Supported workflows :
1. Zenith delays (ZTD) for:
  - the native weather model grid (depends on choice of weather model)
  - a user-defined cube (bounding box, optional: height levels, grid spacing, projection)
2. Slant delays (STD) using full raytracing for:
  - a user-defined cube (bounding box, optional: height levels, grid spacing, projection) + orbit file (Sentinel-1 format currently supported)
3. Slant delays (STD) using projection from Zenith for (NEW):
  - a user-defined cube (bounding box, optional: height levels, grid spacing, projection)  + orbit file (Sentinel-1 format currently supported)
  
### Supported Weather Models
- HRRR (as used in the config file)
- ECMWF - HRES works with an account

### Links to example config files and corresponding output
[Zenith delays on weather model grid nodes at specified height levels](https://github.com/dbekaert/RAiDER/blob/dev/test/scenario_1/raider_example_1.yaml) --> [Output](https://github.com/dbekaert/RAiDER/blob/dev/test/scenario_1/HRRR_tropo_20200101T120000_ztd.nc)
[Slant delays using raytracing on a custom defined cube](https://github.com/dbekaert/RAiDER/blob/dev/test/scenario_3/raider_example_3.yaml), [(Orbit file)](https://github.com/dbekaert/RAiDER/blob/dev/test/scenario_3/S1A_OPER_AUX_POEORB_OPOD_20181203T120749_V20181112T225942_20181114T005942.EOF) --> [Output](https://github.com/dbekaert/RAiDER/blob/dev/test/scenario_3/HRRR_tropo_20181113T230000_ray.nc)
[Slant delays using projection on a custom defined cube](https://github.com/dbekaert/RAiDER/blob/dev/test/scenario_3/raider_example_3_proj.yaml), [(Orbit file)](https://github.com/dbekaert/RAiDER/blob/dev/test/scenario_3/S1A_OPER_AUX_POEORB_OPOD_20181203T120749_V20181112T225942_20181114T005942.EOF) --> [Output](https://github.com/dbekaert/RAiDER/blob/dev/test/scenario_3/HRRR_tropo_20181113T230000_std.nc)

### Not supported:
- Cube input files for azimuth and incidence angles
- NISAR Orbits or orbit object entry point
- differential delays
- Temporal interpolation
